### PR TITLE
Add tests

### PR DESCRIPTION
**Pull request by michabay05**
This PR contains some unit tests for setups using YAML files as well as other sanity checks for various typical and edge case scenarios w/ the sensor and controller.

### CODE REVIEW

File: tests/agent/control/test_controllers.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    result: bool = False
+    if binary:
+        if detected:
+            result = actions == on_see
+        else:
+            result = actions == on_nothing
+    else:
+        result = actions == const_output
+
+    return result
```
Comment (kenblu24) [inline tests/agent/control/test_controllers.py:57]: Is this supposed to be an assert? Or is this a pytest feature that requires `result is True`?

File: tests/agent/test_agentconfs.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            collides=True,
+            sensors=[sensor],
+            team="hunter"
+        )
+        self.agent = Agent(self.bac, world)
+
+    def test_position(self):
+        assert len(self.agent.pos) == len(self.bac.position)
+        assert self.agent.pos[0] == self.bac.position[0]
+        assert self.agent.pos[1] == self.bac.position[1]
```
[the remark this answers is not in the document]
Comment (michabay05) [reply tests/agent/test_agentconfs.py:38]: I agree with this one. When I originally wrote it, I was not sure how python handled list comparison.

File: tests/world/test_out_of_bounds.py
[the remark this answers is not in the document]
Comment (michabay05) [reply tests/world/test_out_of_bounds.py:1]: Yep exactly.

File: tests/agent/control/test_controllers.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (michabay05) [reply tests/agent/control/test_controllers.py:57]: Yes, it should be an assert. I forgot to change it once I added the pytest "parameterization" stuff w/ yaml file "globbing".

File: tests/agent/test_agentconfs.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+class TestAgentConf:
+    @pytest.fixture(autouse=True)
+    def agent_conf_setup(self, mocker: MockerFixture):
+        world = mocker.Mock(spec=RectangularWorld)
+        world.rng = np.random.default_rng()
+        world.dt = 1 / 40
+        controller = mocker.Mock(spec=BinaryController)
+        controller.agent = None
+        sensor = mocker.Mock(spec=BinaryFOVSensor)
```
[the remark this answers is not in the document]
Comment (michabay05) [reply tests/agent/test_agentconfs.py:20]: The intention for using `Mock` here is to not worry about providing exact parameter values for the sensor and controller. Among all the uses of `Mock` in this file, these two are easiest to replace without much consequence; so, replacing them with actual instances with exact parameters would work just as fine for this context.

**Comment by kenblu24** [comment]
You may also be interested to know that Numpy has some really useful methods for testing if array elements match:

* [`np.array_equal()`](https://numpy.org/doc/stable/reference/generated/numpy.array_equal.html) Returns True if both arrays have same shape, and they are elementwise equal.
* [`np.array_equiv()`](https://numpy.org/doc/stable/reference/generated/numpy.array_equiv.html) Returns True if they are elementwise equal, and the shape can be broadcast to be the same.
* [`np.allclose()`](https://numpy.org/doc/stable/reference/generated/numpy.allclose.html) Broadcasts the arrays, and returns True if all values are close to each other within a specified tolerance. Note that there are two types of tolerance you can specify.
* [`np.isclose()`](https://numpy.org/doc/stable/reference/generated/numpy.isclose.html) Does the same as `allclose` except returns a boolean _array_, so you have info on *which* elements are close.

The [difference between](https://www.skytowner.com/explore/difference_between_the_methods_array_equal_and_array_equiv_in_numpy) `np.array_equal()` and `np.array_equiv()` is a bit subtle, so if in doubt use the former.
